src/tardev-snapshotter/src/snapshotter.rs

```diff
@@ -1012,16 +1012,79 @@ impl TarDevSnapshotter {
                         return Err(anyhow!(copy_error));
                     }
                 }
+                file.flush()?;
```

Comment:
Should we close the file too? (not familiar from top of my head with the rust file api)

Reply:
Used drop(file) to close the file